### PR TITLE
Add the formula for biogeme

### PR DESCRIPTION
**Pull request by michelbierlaire**
- [YES] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [YES] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [YES] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [YES] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/biogeme.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # This test will fail and we won't accept that! It's enough to just replace
+    # "false" with the main program this formula installs, but it'd be nice if you
+    # were more thorough. Run the test with `brew test biogeme`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
+    (testpath/"minimal.py").write("from biogeme import *\nrowIterator('obsIter')\nBIOGEME_OBJECT.SIMULATE = Enumerate({'Test':1},'obsIter')")
```
Comment (bfontaine) [inline Formula/biogeme.rb]: Please use `<<-EOS.undent` if the content is too long to fit on one line:

```rb
(testpath/"minimal.py").write <<-EOS.undent
  from biogeme import *
  rowIterator('obsIter')
  # ...
EOS
```

File: Formula/biogeme.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  test do
+    # `test do` will create, run in and delete a temporary directory.
+    #
+    # This test will fail and we won't accept that! It's enough to just replace
+    # "false" with the main program this formula installs, but it'd be nice if you
+    # were more thorough. Run the test with `brew test biogeme`. Options passed
+    # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
+    #
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
```
Comment (bfontaine) [inline Formula/biogeme.rb]: Please remove these comments.

File: Formula/biogeme.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
+class Biogeme < Formula
+  desc "Maximum likelihood estimation of choice models"
+  homepage "http://biogeme.epfl.ch"
+  url "http://biogeme.epfl.ch/distrib/biogeme-2.5.tar.gz"
+  sha256 "88548e99f4f83c24bf7ddb8e0de07588adc2bec515569c56e816ed5b20a624b3"
+
+  option "with-bison", "Install also Bison Biogeme"
```
Comment (bfontaine) [inline Formula/biogeme.rb]: Is there a reason not to install it?

**Review by bfontaine — CHANGES_REQUESTED** [review]

**Comment by bfontaine** [comment]
> [YES] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?

The guidelines include our preferred commit message format for new formulae:

> * `git commit` with message formatted `foo 2.3.4 (new formula)`

In this case it should be `biogeme 2.5 (new formula)`.

**Comment by michelbierlaire** [comment]
I think that I have been able to change and commit the message. Is it OK now? Thanks.

File: Formula/biogeme.rb (hunk 3 above)
Comment (michelbierlaire) [inline Formula/biogeme.rb]: Yes. It is a version that will be abandoned in the near future.

File: Formula/biogeme.rb (hunk 2 above)
Comment (michelbierlaire) [inline Formula/biogeme.rb]: Done.

File: Formula/biogeme.rb (hunk 1 above)
Comment (michelbierlaire) [inline Formula/biogeme.rb]: Done.

**Review by bfontaine — APPROVED** [review]

File: Formula/biogeme.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  test do
+    (testpath/"minimal.py").write <<-EOS.undent
+    from biogeme import *
+    rowIterator('obsIter')
+    BIOGEME_OBJECT.SIMULATE = Enumerate({'Test':1},'obsIter')
+    EOS
+    (testpath/"minimal.dat").write <<-EOS.undent
+    TEST
+    1
+    EOS
```
Comment (bfontaine) [inline Formula/biogeme.rb]: Please indent each file content by two spaces:

```rb
# before
(testpath/"minimal.dat").write <<-EOS.undent
TEST
1
EOS

# after
(testpath/"minimal.dat").write <<-EOS.undent
  TEST
  1
EOS
```

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/biogeme.rb (hunk 3 above)
Comment (bfontaine) [inline Formula/biogeme.rb]: So… is there a reason to install it? :D 
We try to limit the number of options provided by a formula to its minimum. I’m just checking if this option is necessary.

Comment (michelbierlaire) [inline Formula/biogeme.rb]: Good point. I have removed all options.

File: Formula/biogeme.rb (hunk 4 above)
Comment (michelbierlaire) [inline Formula/biogeme.rb]: Done.

**Review by bfontaine — APPROVED** [review]

File: Formula/biogeme.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  def install
+    args = %W[
+      --prefix=#{prefix}
+    ]
+    args << "--enable-python"
+    args << "--enable-bison"
+    args << "--enable-gui"
+
+    system "./configure", *args
```
Comment (bfontaine) [inline Formula/biogeme.rb]: Now that this array is always the same, you can inline the arguments:
```rb
system "./configure", "--prefix=#{prefix}",
                      "--enable-python",
                      "--enable-bison",
                      "--enable-gui"
```

Comment (michelbierlaire) [inline Formula/biogeme.rb]: Done.

**Comment by michelbierlaire** [comment]
Merci beaucoup ! Je comprends pourquoi homebrew fonctionne si bien. Le processus de review est très sérieux. Bravo.

**Comment by michelbierlaire** [comment]
It is my first contribution to github. Do I need to take any other action, now that the added formula has been approved?

**Comment by fxcoudert** [comment]
@michelbierlaire merci !

Ideally, you would now squash all your commits into a single one, with message `biogeme 2.5 (new formula)`. Then we can merge it.

**Comment by michelbierlaire** [comment]
I am not sure to understand what you mean by "squash all your commits". When I try to commit again, it says "nothing to commit, working tree clean"

**Comment by fxcoudert** [comment]
I meant rebase your branch's history so that instead of 6 commits, there is only 1. This makes less work for us to merge into a single commit. But I will take care of that now.

**Comment by fxcoudert** [comment]
Thanks @michelbierlaire for your contribution to Homebrew! We look forward to you contributing again.

**Comment by bfontaine** [comment]
Thanks @fxcoudert for merging this :smiley:

**Pull request closed without merging**